--- origsrc/lua-gtk-0.7/script/Makefile.common	2008-01-11 10:28:26.000000000 -0600
+++ src/lua-gtk-0.7/script/Makefile.common	2008-02-14 00:33:03.765625000 -0600
@@ -201,7 +201,7 @@
 	$(LIBINSTALL) ${ODIR}${ODLL} $(INDIR1)/${ODLL}
 	$(LIBFINAL) $(INDIR1)
 	mkdir -p $(INDIR2)/gtk
-	cp -a lib $(INDIR2)/gtk/
+	cp -L lib/* $(INDIR2)/gtk/
 
 tar:
 	(cd ..; ln -s lua-gtk lua-gtk-${VERSION}; \
--- origsrc/lua-gtk-0.7/script/Makefile.cygwin	1969-12-31 18:00:00.000000000 -0600
+++ src/lua-gtk-0.7/script/Makefile.cygwin	2008-02-14 00:39:03.906250000 -0600
@@ -0,0 +1,31 @@
+# Makefile for lua-gtk2 (Linux)
+
+# --- begin platform configuration ---
+PLATFORM:=CYGWIN
+LIBFFI	:=/usr/lib/libffi.a
+
+# if you don't have cmph (http://sf.net/projects/cmph/), just comment this out.
+#CMPHDIR	:=/build/cmph-0.6/src
+#CMPHBIN	:=cmph
+#CMPHLIB	:=-lcmph -lm
+
+# Installation into these directories.
+INDIR1	:=$(DESTDIR)/usr/lib/lua/5.1
+INDIR2	:=$(DESTDIR)/usr/share/lua/5.1
+
+# you probably don't want to change these...
+LD		:=gcc
+LINK	:=0
+MINGW	:=
+ODIR	:=build/cygwin/
+# -fPIC required for position indendent code in shared objects; besides,
+#  shrinks the library size somewhat.  For production build, -O2 is probably ok
+CFLAGS	:=-DPIC
+GTKLIB	:=`pkg-config --libs gtk+-2.0`
+LUALIB	:=-llua5.1
+ODLL	:=gtk.dll
+# --- end platform configuration ---
+
+all: dirs gtk2dll
+
+include script/Makefile.common
--- origsrc/lua-gtk-0.7/src/callback.c	2008-01-09 15:02:40.000000000 -0600
+++ src/lua-gtk-0.7/src/callback.c	2008-02-14 00:18:38.234375000 -0600
@@ -22,7 +22,7 @@
 #include "luagtk_ffi.h"
 
 // use this for older FFI versions doesn't detect existing functions!
-// #define ffi_closure_alloc(x,y) g_malloc(x)
+#define ffi_closure_alloc(x,y) g_malloc(x)
 
 /* one such structure per connected callback */
 struct callback_info {
--- origsrc/lua-gtk-0.7/src/data.c	2008-01-11 09:32:59.000000000 -0600
+++ src/lua-gtk-0.7/src/data.c	2008-02-14 00:29:03.906250000 -0600
@@ -30,6 +30,14 @@
 #ifdef LINUX
     "/usr/lib/libgtk-x11-2.0.so",
 #endif
+#ifdef CYGWIN
+    "cyggtk-x11-2.0-0.dll",
+    "cyggdk-x11-2.0-0.dll",
+    "cyggdk_pixbuf-2.0-0.dll",
+    "cygpango-1.0-0.dll",
+    "cyggobject-2.0-0.dll",
+    "cygglib-2.0-0.dll",
+#endif
 #ifdef WIN32
     "libgtk-win32-2.0-0.dll",
     "libgdk-win32-2.0-0.dll",
@@ -261,7 +269,7 @@
     return p;
 }
 
-#ifdef LINUX
+#if defined LINUX || defined CYGWIN
 void *dl_handle[MAX_DLL];
 #endif
 
--- origsrc/lua-gtk-0.7/src/luagtk.h	2007-12-20 15:23:57.000000000 -0600
+++ src/lua-gtk-0.7/src/luagtk.h	2008-02-14 00:10:49.578125000 -0600
@@ -2,7 +2,7 @@
 
 #include "config.h"
 
-#ifdef LINUX
+#if defined LINUX || defined CYGWIN
 #include <dlfcn.h>
 #include <ffi.h>		/* foreign function interface library */
 #define EXPORT
